src/main/java/com/example/demo/services/impl/UserServiceImpl.java

```diff
@@ -52,19 +82,29 @@ public User getUserById(final long id) {
 
     @Override
     public void addUser(final UserCreation user) {
+        ageValidation(user.getAge());
         userRepository.save(userEntityMapper.map(user));
     }
 
     @Override
     public void updateUser(final UserUpdate userUpdate) {
-        final LocalDateTime creationDate = userRepository.getById(userUpdate.getId()).getCreatedAt();
-        final UserEntity updatedUserEntity = userEntityMapper.map(userUpdate);
-        updatedUserEntity.setCreatedAt(creationDate);
+        ageValidation(userUpdate.getAge(), userUpdate.getId());
+
+        final UserEntity updatedUserEntity = userRepository.getById(userUpdate.getId());
+        updatedUserEntity.setName(userUpdate.getName());
+        updatedUserEntity.setAge(userUpdate.getAge());
+        updatedUserEntity.setRole(userUpdate.getRole());
+        updatedUserEntity.setPassword(userUpdate.getPassword());
         userRepository.save(updatedUserEntity);
     }
 
     @Override
     public void deleteUser(final long id) {
+        if (!userRepository.existsById(id)) {
+            log.error(USER_DELETE_ERROR);
+            throw new RuntimeException(USER_DELETE_ERROR);
+        }
         userRepository.deleteById(id);
+        log.info(USER_DELETE_INFO  + ", user id: " + id);
```

Comment:
Use this format:
 `private static final String USER_DELETE_INFO  = "User with id %s has been deleted";`
...
 `log.info(format(USER_DELETE_INFO, id))`
 
 P.S. format is from import static java.lang.String.format;